File: backend/scripts/load_symptoms.py
import os
import pandas as pd
from sqlalchemy import create_engine, text

def main():
    raw = os.getenv("DATABASE_URL", "").strip()
    if raw.startswith("postgres://"):
        raw = raw.replace("postgres://", "postgresql://", 1)
    if not raw:
        raw = "postgresql+psycopg2://pharmacy_user:1234@localhost:5432/pharmacy"
    engine = create_engine(raw)
    print("→ Connecting to:", raw)

    csv_path = os.path.join(os.path.dirname(__file__), "../all_symptoms.csv")
    if not os.path.exists(csv_path):
        raise FileNotFoundError(f"Could not find symptoms CSV at {csv_path}")
    df = pd.read_csv(csv_path, usecols=["name"])
    df["name"] = df["name"].str.strip().str.lower()
    df = df.dropna().drop_duplicates()

    print(f"→ Loaded {len(df)} unique symptom names from CSV")

    with engine.begin() as conn:
        conn.execute(text("DROP TABLE IF EXISTS tmp_symptoms"))
        conn.execute(text("CREATE TEMP TABLE tmp_symptoms(name TEXT)"))

        df.to_sql("tmp_symptoms", conn, if_exists="append", index=False, method="multi")
        print("→ Inserted into tmp_symptoms")

        stmt = """
        INSERT INTO symptom_lookup(name)
          SELECT name FROM tmp_symptoms
        ON CONFLICT (name) DO NOTHING;
        """
        result = conn.execute(text(stmt))
        print(f"Upserted {result.rowcount} new symptoms into symptom_lookup")

    print("All done.")

if __name__ == "__main__":
    main()

File: backend/scripts/load_symptoms.js


const path = require("path");
const fs   = require("fs");
const { Pool } = require("pg");

async function main() {
  const csvPath = path.resolve(__dirname, "D:pharma/backend/all_symptoms.csv");
  if (!fs.existsSync(csvPath)) {
    console.error(`${csvPath}`);
    process.exit(1);
  }

  const pool = new Pool({
    connectionString:
      process.env.DATABASE_URL ||
      "postgres://pharmacy_user:1234@localhost:5432/pharmacy",
  });

  const client = await pool.connect();
  try {

    await client.query(`DROP TABLE IF EXISTS tmp_symptoms;`);
    await client.query(`CREATE TEMP TABLE tmp_symptoms(name text);`);

    const copyFrom = `COPY tmp_symptoms(name)
                      FROM STDIN WITH (FORMAT csv, HEADER true)`;
    const stream = client.query(copyFrom);
    const fileStream = fs.createReadStream(csvPath);
    fileStream.pipe(stream);
    await new Promise((resolve, reject) => {
      stream.on("end", resolve);
      stream.on("error", reject);
    });

    await client.query(`
      INSERT INTO symptom_lookup(name)
        SELECT DISTINCT name
          FROM tmp_symptoms
      ON CONFLICT (name) DO NOTHING
    `);

    console.log("Loaded symptoms into symptom_lookup");
  } catch (err) {
    console.error(" Failed:", err);
  } finally {
    client.release();
    await pool.end();
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
